jmetal-core/src/main/java/org/uma/jmetal/qualityindicator/impl/Epsilon.java

```diff
@@ -85,29 +85,42 @@ private double epsilon(Front front, Front referenceFront) throws JMetalException
 
     for (int i = 0; i < referenceFront.getNumberOfPoints(); i++) {
       for (int j = 0; j < front.getNumberOfPoints(); j++) {
-        for (int k = 0; k < numberOfObjectives; k++) {
-          epsTemp = front.getPoint(j).getDimensionValue(k)
-              - referenceFront.getPoint(i).getDimensionValue(k);
-          if (k == 0) {
-            epsK = epsTemp;
-          } else if (epsK < epsTemp) {
-            epsK = epsTemp;
-          }
-        }
-        if (j == 0) {
-          epsJ = epsK;
-        } else if (epsJ > epsK) {
-          epsJ = epsK;
-        }
+        epsK = numberofobjectives(front, referenceFront, epsK, numberOfObjectives, i, j);
+        epsJ = jzequalzero(epsJ, epsK, j);
       }
-      if (i == 0) {
+      eps = iequalzero(eps, epsJ, i);
+    }
+    return eps;
```

Comment:
There is clearly something wrong here. There is 3 levels, and we arbitrarily get rid of the 3rd one to create dedicated methods... The point is NOT to blindly cut just to decrease cognitive complexity. By the way, cognitive complexity HAVE NOT BEEN REDUCED BUT INCREASED! Cognition is a human trait, not a machine one. The whole point of this indicator is to simplify the understanding of the code. Can someone explain me how moving a formula to compute an epsilon into a function named `numberofobjectives` is clearer? And what is that `iequalzero` which takes `eps` and `epsJ`? And `jz`... seriously? Do you want my `jz` that much?

This is CODE OBFUSCATION!!! Who is the guy who believed that he could just apply some tool to automatically make the code clearer to humans? These tools can *help* to refactor, but **they don't do the job**!